### ISSUE TITLE
arm build

### ISSUE THREAD

**Issue by nwg-piotr** [dlasky/gotk3-layershell#6]
Hi,

Somewhat out of boredom, I'm trying to build nwg-drawer for ARM (and run it on my archaic Asus C201PA Chromebook). I've never tried to build anything for other architecture. That's what I'm trying, and the result:

```text
$ env GOOS=linux GOARCH=arm go build -o nwg-drawer_arm .
package github.com/nwg-piotr/nwg-drawer
	imports github.com/dlasky/gotk3-layershell/layershell: build constraints exclude all Go files in /home/piotr/go/pkg/mod/github.com/dlasky/gotk3-layershell@v0.0.0-20210827021656-e6ecab2731f7/layershell
```

Do you by any chance have an idea on how to fix it?

**Comment by dlasky** [comment]
Remove the trailing dot I think

**Comment by nwg-piotr** [comment]
The same result.

**Comment by dlasky** [comment]
oh maybe ENABLE_CGO=1 is the issue? like its not allowing the cgo based stuff based on the constraints? its a weird error. I thought cgo was enabled by default though, maybe not cross arch? are you trying to compile this on x86?

**Comment by nwg-piotr** [comment]
> are you trying to compile this on x86?

Yes, I couldn't find a go version for armv7.

This is nothing serious, neither urgent. I've just tried to give my no-longer-supported Chromebook a second life. Actually it works decently on sway, even if installed on SD card.

[edit]
> oh maybe ENABLE_CGO=1 is the issue?

I'll give it a try tonight.

**Comment by dlasky** [comment]
Yeah I saw the Reddit post. I'm glad we have the same hobby -- installing Linux on things lol

**Comment by nwg-piotr** [comment]
Turns out that [armv6l version](https://golang.org/dl/go1.17.3.linux-armv6l.tar.gz) works perfectly well on armv7, and I managed to build nwg-drawer on the target machine. I only had to build gtk-layer-shell from source, since the keyboard support is missing from the version 5.2.2 version, available on Debian. Building took ages, but it looks great.

![20211117_03h32m35s_grim](https://user-images.githubusercontent.com/20579136/142111121-418e6fe1-6f68-44d2-b51b-ee2ff2cef0d0.png)
